### PR TITLE
Update process around merging PRs

### PR DESCRIPTION
**Pull request by asfalcone**
### Description

If you review a PR that is submitted by someone with a `Member` tag, they should merge it themselves. Otherwise that PR should be merged for them.
image

### Motivation and Context

Codify our process around merging PRs

### CODE REVIEW

**Review by KrauseFx — APPROVED** [review]

File: CORE_CONTRIBUTOR.md
Comment (AliSoftware) [inline CORE_CONTRIBUTOR.md]: I believe the term GitHub uses is "Push access" rather than "Commit access"?

Maybe describe the reason from the intent (= "don't merge for them… because they have the necessary rights to merge it themselves") instead of the consequence (commit access)?

> Merge pull requests we review, except for PRs where the author themselves has the right to merge them. Github annotate those authors with the `Member` tag

Comment (AliSoftware) [inline CORE_CONTRIBUTOR.md]: > Merge pull requests we review, except for PRs where the author themselves is a core contributor. Github annotate people in the Core Contributors team with the `Member` tag

**Review by AliSoftware — APPROVED** [review]

**Review by AliSoftware — DISMISSED** [review]
Another wording suggestion, to also talk about using the Review changes feature integrated in GitHub:

> - Once we agree with the content of a PR:
>   - Merge the PRs if the author doesn't have push access to merge it themselves.
>   - Just mark the request as Approved For PRs authored by someone who has push access, and let them merge their PR themselves.
>   - ℹ️ An author with push access to _fastlane_ repos is noted in Github with the `Member` tag

**Comment by AliSoftware** [comment]
Add a mention about using "Squash and Merge" instead of "Merge"

**Pull request merged**
